test/metrics/metrics.ts

```diff
@@ -0,0 +1,407 @@
+        assert.strictEqual(gfeLatency, 123);
+      });
+
+      // check that the latency matches up with the measured elapsed time within 10ms
+      assertApprox(elapsedTime, totalOperationLatency, 10);
+
+      // Make sure no GFE connectivity errors ar emitted since we got GFE latencies
+      const gfeMissingData = hasMetricData(
+        resourceMetrics,
+        METRIC_NAME_GFE_CONNECTIVITY_ERROR_COUNT,
+      );
+
+      assert.ok(!gfeMissingData);
+
+      await database.close();
+    });
+
+    it('should increase attempts on retries', async () => {
+      gfeStub = sandbox
+        .stub(MetricsTracer.prototype, 'extractGfeLatency')
+        .callsFake((header: string) => 123);
+      const database = newTestDatabase();
+      const err = {
+        message: 'Temporary unavailable',
+        code: grpc.status.UNAVAILABLE,
+      } as MockError;
+      spannerMock.setExecutionTime(
+        spannerMock.executeStreamingSql,
+        SimulatedExecutionTime.ofError(err),
+      );
+
+      await database.run(selectSql);
+      const {resourceMetrics} = await reader.collect();
+
+      const operationCountData = getMetricData(
+        resourceMetrics,
+        METRIC_NAME_OPERATION_COUNT,
+      );
+      const attemptCountData = getMetricData(
+        resourceMetrics,
+        METRIC_NAME_ATTEMPT_COUNT,
+      );
+      const operationLatenciesData = getMetricData(
+        resourceMetrics,
+        METRIC_NAME_OPERATION_LATENCIES,
+      );
+      const attemptLatenciesData = getMetricData(
+        resourceMetrics,
+        METRIC_NAME_ATTEMPT_LATENCIES,
+      );
+      const gfeLatenciesData = getMetricData(
+        resourceMetrics,
+        METRIC_NAME_GFE_LATENCIES,
+      );
+
+      const sessionAttributes = {
+        ...commonAttributes,
+        database: `database-${dbCounter}`,
+        method: 'batchCreateSessions',
+      };
+      // Verify batchCreateSession metrics are unaffected
+      assert.strictEqual(
+        getAggregatedValue(operationCountData, sessionAttributes),
+        1,
+      );
+      getAggregatedValue(operationLatenciesData, sessionAttributes);
+      assert.strictEqual(
+        getAggregatedValue(attemptCountData, sessionAttributes),
+        1,
+      );
+      getAggregatedValue(attemptLatenciesData, sessionAttributes);
+      assert.strictEqual(
+        getAggregatedValue(gfeLatenciesData, sessionAttributes),
+        123,
+      );
+
+      const executeAttributes = {
+        ...commonAttributes,
+        database: `database-${dbCounter}`,
+        method: 'executeStreamingSql',
+      };
+      // Verify executeStreamingSql has 2 attempts and 1 operation
+      assert.strictEqual(
+        getAggregatedValue(operationCountData, executeAttributes),
+        1,
+      );
+      getAggregatedValue(operationLatenciesData, executeAttributes);
+      assert.strictEqual(
+        getAggregatedValue(attemptCountData, executeAttributes),
+        2,
+      );
```

Comment:
This is not correct. Currently attemptCount is 2, but both are having OK status. 

It should be 1 attempt with "UNAVAILABLE" status and other with "OK"